### PR TITLE
images: Move hubble-proto into cilium-builder

### PR DESCRIPTION
**Pull request by gandro**
This commit consolidates the `hubble-proto` image (used to build the
Hubble protobuf definitions) and the `cilium-builder` image. While this
increases the download size for developers just needing to regenerate
the Hubble API definitions, removing `hubble-proto` reduces the
maintenance overhead of our developer images (for example, it currently
lacks the required GitHub Action to build and push it to docker.io).

### CODE REVIEW

File: api/v1/Makefile [hunk 1]
```diff
@@ -1,7 +1,14 @@
...
+CILIUM_BUILDER_IMAGE=quay.io/cilium/cilium-builder:3e04d507e2f7a0b569490060ddc8b9423c9fc8f0@sha256:7949a297fd2f07a008ef8a163ed84bfb338a37f3b36b68c2c9c5fddb324d325f
 
 .PHONY: proto
 proto:
-	docker container run --rm \
-		--volume $(CURDIR):/source \
-		cilium/hubble-proto:${version}
+	$(QUIET)$(CONTAINER_ENGINE) container run --rm \
+		--volume $(CURDIR):/src \
+		--user "$(shell id -u):$(shell id -g)" \
```
Comment (rolinh) [inline api/v1/Makefile:12]: Thanks for this, running as a regular user is definitely something we want (it was also part of my old draft PR #13405).

**Review by rolinh — APPROVED** [review]
Thanks a lot, LGTM :+1:

**Review by tklauser — APPROVED** [review]
Nice :tada:

**Comment by gandro** [comment]
test-me-please

**Review by glibsm — APPROVED** [review]
Tried locally, worked for me.

**Comment by gandro** [comment]
test-1.19-5.4

**Comment by gandro** [comment]
test-runtime

**Comment by gandro** [comment]
test-1.16-netnext

**Comment by gandro** [comment]
All required tests have passed. Marking as ready for merge. GH Actions seems stuck in the non-required ones, so the bot is unable to add the label itself.

The `This branch is waiting to be deployed` action can also be ignored, we already updated the image before I amended by commits according to https://docs.cilium.io/en/latest/contributing/development/images/#update-cilium-builder-and-cilium-runtime-images

**Pull request merged**